dds_registration/models.py

```diff
@@ -206,34 +221,88 @@ def __str__(self):
         return info
 
 
-class Registration(models.Model):
-    """
-    Ex `Booking` class in OneEvent
-    """
+class Message(models.Model):
+    event = models.ForeignKey(Event, on_delete=models.CASCADE)
+    message = models.TextField()
+    emailed = models.BooleanField(default=False)
 
-    invoice_no = models.AutoField(primary_key=True)
+    created_at = models.DateTimeField(auto_now_add=True)
 
-    event = models.ForeignKey(Event, related_name="registrations", on_delete=models.CASCADE)
-    user = models.ForeignKey(User, related_name="registrations", on_delete=models.CASCADE)
+    def __str__(self):
+        items = [
+            self.event,
+            self.created_at.strftime(dateTimeFormat) if self.created_at else None,
+            "emailed" if self.emailed else None,
+        ]
+        info = ", ".join(filter(None, map(str, items)))
+        return info
 
-    # If the registration has cancelled, the `active` status should be set to false
-    active = models.BooleanField(default=True)
 
-    options = models.ManyToManyField(RegistrationOption)
+class Invoice(models.Model):
+    # Docs recommend putting these on the class:
+    # https://docs.djangoproject.com/en/5.0/ref/models/fields/#django.db.models.Field.choices
+    INVOICE_STATUS = [
+        # Can add other possibilities later
+        ("CREATED", "Created"),
+        ("ISSUED", "Issued"),
+        ("PAID", "Paid"),
+        ("REFUNDED", "Refunded"),
+    ]
+
+    INVOICE_TEMPLATES = [
+        # Different templates for specific bank accounts
+        # and layouts
+        ("M-CHF", "Membership - Swiss Francs"),
+        ("M-EUR", "Membership - Euros"),
+        ("G-USD", "Generic - USD"),
+        ("G-CHF", "Generic - CHF"),
+        ("G-EUR", "Generic - EUR"),
+        ("G-CAD", "Generic - CAD"),
+    ]
 
-    # Payment method:
     PAYMENT_METHODS = [
         ("STRIPE", "Stripe"),
         ("INVOICE", "Invoice"),
+        # Not yet implemented
+        ("WISE", "Wise"),
     ]
-    DEFAULT_PAYMENT_METHOD = "STRIPE"
-    payment_method = models.TextField(choices=PAYMENT_METHODS, default=DEFAULT_PAYMENT_METHOD)
 
-    extra_invoice_text = models.TextField(blank=True, default="")
+    # Same as the actual invoice number, which normally has the form
+    # {two-digit-year}{zero-padded four digit number starting from 1}
+    invoice_no = models.IntegerField(primary_key=True)
+    created = models.DateField(auto_now_add=True)
+    status = models.TextField(choices=INVOICE_STATUS)
+    # Includes the various item descriptions, prices, and currencies
+    # and any other necessary info
+    # The specific form will depend on the template
+    data = models.JSONField()
+    template = models.TextField(choices=INVOICE_TEMPLATES)
+
+
+REGISTRATION_STATUS = [
```

Comment:
Probably, it'd be better to have it as a static member of the `Registration` model, like other choices lists?